squeal-postgresql/src/Squeal/PostgreSQL/Expression/Text.hs

```diff
@@ -63,3 +67,10 @@ like = unsafeBinaryOp "LIKE"
 -- ((E'abc' :: text) ILIKE (E'a%' :: text))
 ilike :: Operator (null 'PGtext) (null 'PGtext) ('Null 'PGbool)
 ilike = unsafeBinaryOp "ILIKE"
+
+-- | Determines the location of the substring match using the `strpos`
+-- function. If `0` is returned, no match, else the index of the match
+-- location is returned.
+strpos
+  :: '[null0 'PGtext, null1 'PGtext] ---> (CombineNullity null0 null1) 'PGint4
+strpos = unsafeFunctionN "strpos"
```

Comment:
This is introducing a new idiom of using `CombineNullity`. The normal way to write this would be
```
strpos
  :: '[null 'PGtext, null 'PGtext] ---> null 'PGint4
```
And force the user to explicitly wrap a non-null arguments with `notNull` (similar to using `Just`) in use it with a null argument. Is there a particular reason to introduce `CombineNullity`?

Reply:
`CombineNullity` is a useful convenience function for working with columns on unknown nullity, but it doesn't really make sense to introduce it here...